src/scripts/cronjobs/emailBreachAlerts/emailBreachAlerts.spec.ts

```diff
@@ -0,0 +1,254 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import type { Logger } from "winston";
+import * as Sentry from "@sentry/node";
+import { mockLogger } from "../../../test/helpers/mockLogger";
+import { mockMessage } from "../../../test/helpers/pubsub";
+import { breachMessageHandler } from "./emailBreachAlerts";
+import { seeds } from "../../../test/db";
+import { createRandomHibpListing as mockBreach } from "../../../apiMocks/mockData";
+import { HibpLikeDbBreach } from "../../../utils/hibp";
+
+const mockSubscriber = seeds.breachNotificationSubscriber;
+
+const defaultBreachName = "ExampleBreach";
+
+const validMsgPayload = {
+  breachName: defaultBreachName,
+  hashPrefix: "AA",
+  hashSuffixes: ["11"],
+};
+
+const validBreachDefaults: Partial<HibpLikeDbBreach> = {
+  Name: defaultBreachName,
+  IsFabricated: false,
+  IsSpamList: false,
+  IsVerified: true,
+  Domain: "http://example.com",
+};
+
+describe("breachMessageHandler", () => {
+  let logger: Logger;
+  const breadcrumbSpy = jest.spyOn(Sentry, "addBreadcrumb").mockReturnValue();
+  const setTagSpy = jest.spyOn(Sentry, "setTag").mockReturnValue();
+
+  beforeEach(() => {
+    logger = mockLogger() as unknown as Logger;
+    jest.clearAllMocks();
+  });
+  afterAll(() => {
+    jest.restoreAllMocks();
+  });
+
+  it("succeeds and notifies only eligible recipients", async () => {
+    // All types of notifications, 1 disabled
+    const recipients = [false, true, null].map((override) =>
+      mockSubscriber({ all_emails_to_primary: override }),
+    );
+    const breachProvider = async () => [mockBreach(validBreachDefaults)];
+    // Set up mocked injected dependencies
+    const subs = { findByHashes: jest.fn().mockResolvedValue(recipients) };
+    const notifications = {
+      subscriberNotifiedForBreach: jest.fn().mockResolvedValue(false),
+      addEmailNotification: jest.fn().mockResolvedValue(undefined),
+      markEmailAsNotified: jest.fn().mockResolvedValue(undefined),
+    };
+    const sendEmail = jest.fn().mockResolvedValue(undefined);
+
+    const message = mockMessage(validMsgPayload);
+
+    const res = await breachMessageHandler(
+      message,
+      logger,
+      breachProvider,
+      subs,
+      notifications,
+      sendEmail,
+    );
+
+    expect(res).toEqual({ success: true, errors: 0, notified: 2, skipped: 1 });
+    expect(subs.findByHashes).toHaveBeenCalledTimes(1);
+    expect(sendEmail).toHaveBeenCalledTimes(2);
+    expect(notifications.addEmailNotification).toHaveBeenCalledTimes(2);
+    expect(notifications.markEmailAsNotified).toHaveBeenCalledTimes(2);
+  });
+
+  it("skips if breach is not notifiable", async () => {
+    const breach = mockBreach({ Name: defaultBreachName, IsVerified: false }); // not notifiable
```

Comment:
Suggestion: to be thorough, also check on other reasons for not notifying? (empty domains, fabricated, spam)